model/task_start_estimation.go

```diff
@@ -0,0 +1,150 @@
+package model
+
+import (
+	"sort"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const (
+	HostInitializingDelay  = 4 * time.Minute
+	HostStartingDelay      = 3 * time.Minute
+	HostProvisiongingDelay = 1 * time.Minute
+)
+
+type estimatedTask struct {
+	duration time.Duration
+}
+
+type estimatedHost struct {
+	timeToCompletion time.Duration
+}
```

Comment:
why not just store slices of durations rather than these types? (which you could write methods on)

Reply:
I wanted to make these structs in case we needed to add more factors into the estimation in the future

Reply:
Like what? Is it that hard to change it to structs if/when we need to add more inputs?